internal/remote/modulereleasemeta_syncworker.go

```diff
@@ -0,0 +1,170 @@
+
+// moduleReleaseMetaConcurrentWorker performs ModuleReleaseMeta synchronization using multiple goroutines.
+type moduleReleaseMetaConcurrentWorker struct {
+	namespace  string
+	patchDiff  func(ctx context.Context, obj *v1beta2.ModuleReleaseMeta) error
+	deleteDiff func(ctx context.Context, obj *v1beta2.ModuleReleaseMeta) error
+	createCRD  func(ctx context.Context) error
+}
+
+// newModuleReleaseMetaConcurrentWorker returns a new moduleReleaseMetaConcurrentWorker instance with default dependencies.
+func newModuleReleaseMetaConcurrentWorker(kcpClient, skrClient client.Client, settings *Settings) *moduleReleaseMetaConcurrentWorker {
+	patchDiffFn := func(ctx context.Context, obj *v1beta2.ModuleReleaseMeta) error {
+		return patchDiffModuleReleaseMeta(ctx, obj, skrClient, settings.SSAPatchOptions)
+	}
+
+	deleteDiffFn := func(ctx context.Context, obj *v1beta2.ModuleReleaseMeta) error {
+		return deleteModuleReleaseMeta(ctx, obj, skrClient)
+	}
+
+	createCRDFn := func(ctx context.Context) error {
+		return createModuleReleaseMetaCRDInRuntime(ctx, kcpClient, skrClient)
+	}
+
+	return &moduleReleaseMetaConcurrentWorker{
+		namespace:  settings.Namespace,
+		patchDiff:  patchDiffFn,
+		deleteDiff: deleteDiffFn,
+		createCRD:  createCRDFn,
+	}
+}
+
+// SyncConcurrently synchronizes ModuleReleaseMetas from KCP to SKR.
+// kcpModules are the ModuleReleaseMetas to be synced from the KCP cluster.
+func (c *moduleReleaseMetaConcurrentWorker) SyncConcurrently(ctx context.Context, kcpModules []v1beta2.ModuleReleaseMeta) error {
+	channelLength := len(kcpModules)
+	results := make(chan error, channelLength)
+	for kcpIndex := range kcpModules {
+		go func() {
+			prepareModuleReleaseMetaForSSA(&kcpModules[kcpIndex], c.namespace)
+			results <- c.patchDiff(ctx, &kcpModules[kcpIndex])
+		}()
+	}
+	var errs []error
+	for range channelLength {
+		if err := <-results; err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	// retry if ModuleReleaseMeta CRD is not existing in SKR cluster
+	if containsCRDNotFoundError(errs) {
+		if err := c.createCRD(ctx); err != nil {
+			return err
+		}
+	}
+
+	if len(errs) != 0 {
+		errs = append(errs, errCatModuleReleaseMetaApply)
+		return errors.Join(errs...)
+	}
+	return nil
+}
+
+// DeleteConcurrently deletes ModuleReleaseMetas from SKR.
+func (c *moduleReleaseMetaConcurrentWorker) DeleteConcurrently(ctx context.Context,
+	diffsToDelete []v1beta2.ModuleReleaseMeta,
+) error {
+	channelLength := len(diffsToDelete)
+	results := make(chan error, channelLength)
+	for _, diff := range diffsToDelete {
+		go func() {
+			results <- c.deleteDiff(ctx, &diff)
+		}()
+	}
+	var errs []error
+	for range channelLength {
+		if err := <-results; err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	if len(errs) != 0 {
+		errs = append(errs, errModuleReleaseMetaCleanup)
+		return errors.Join(errs...)
+	}
+	return nil
+}
+
+func createModuleReleaseMetaCRDInRuntime(ctx context.Context, kcpClient client.Client, skrClient client.Client) error {
```

Comment:
This function shares the same logic with createModuleTemplateCRDInRuntime, so we can move it to the crd.go and just pass the crdType (ModuleTemplateKind or ModuleReleaseMetaKind).